src/Core/Configure/FileConfigTrait.php

```diff
@@ -46,14 +46,7 @@ protected function _getFilePath($key, $checkExists = false)
         }
 
         list($plugin, $key) = pluginSplit($key);
-
-        if ($plugin) {
-            $file = Plugin::configPath($plugin) . $key;
-        } else {
-            $file = $this->_path . $key;
-        }
-
-        $file .= $this->_extension;
+        $file = ($plugin ? Plugin::configPath($plugin) : $this->_path) . $key . $this->_extension;
```

Comment:
Using ternary and concatenation in single statement is not good for readability. Keep the concatenation separate.